core/src/main/java/org/apache/iceberg/ReachableFileUtil.java

```diff
@@ -148,12 +148,49 @@ public static List<String> statisticsFilesLocations(Table table) {
    * @param table table for which statistics files needs to be listed
    * @param predicate predicate for filtering the statistics files
    * @return the location of statistics files
+   * @deprecated use the {@code statisticsFilesLocationsForSnapshots(table, snapshotIds)} instead.
    */
+  @Deprecated
```

Comment:
can't use the `Predicate<StatisticsFile>` for partition stats. Hence, deprecated and introduced snapshot id based input like other methods in this class (manifest list, manifest etc)